### ISSUE TITLE
Split filtering into its own command?

### ISSUE THREAD

**Issue by benjjneb** [qiime2/q2-dada2#51]
As I understand it, QIIME intends to be more of a push-button pipeline than our R workflows, but I think it would be worth considering separating the filtering in the DADA2 pipeline from the sample processing. It is often useful to filter more than once to see what works well, and I would imagine that there will be other filtering tools that will come online in the QIIME2 ecosystem that people may want to use. It also has a nice effect of reducing the number of parameters/options at each step, and naturally grouping them together.

Downside: Two commands to do what once took one. May need a new semantic type (eg. FilteredVersionOfPreviousType).

**Comment by gregcaporaso** [comment]
I agree that it would make sense to do this. At this stage it makes sense to focus on making functionality modular. When we have `Pipeline` actions in the future, we can chain the more modular methods together for users who want the more "push button" functionality.

**Comment by benjjneb** [comment]
Is there a github issue somewhere that I can follow the progress of Q2 Pipelines?

**Comment by jairideout** [comment]
Yes! @ebolyen is actually working on that right now, and the plan is to have basic pipeline support in place for this release cycle (2017.10). Here's the issue and corresponding pull request:

https://github.com/qiime2/qiime2/issues/86
https://github.com/qiime2/qiime2/pull/348

**Comment by ebolyen** [comment]
Pipelines are now merged!

**Comment by thermokarst** [comment]
It might be nice to include the `plotErrors` visualization as part of this process (of course this would require an all new viz in the plugin, but it could wrap the existing `plotErrors` results). This recently [came up on the forum](https://forum.qiime2.org/t/dada2-self-consistency-loop-terminated-before-convergence/1716?u=thermokarst).

**Comment by benjjneb** [comment]
Is there an example Pipeline already implemented, or developer documentation on implementing Pipelines?

**Comment by ebolyen** [comment]
@benjjneb, the [framework section in the release notes](https://forum.qiime2.org/t/qiime-2-2017-10-release-is-now-live/1626) has some light documentation. And the `core-metrics` and `core-metrics-phylogenetic` actions are pipelines:
  - [implementation](https://github.com/qiime2/q2-diversity/blob/d026613e02a574ed0b95506be0976c3fe49e96d4/q2_diversity/_core_metrics.py)
  - [registration](https://github.com/qiime2/q2-diversity/blob/d026613e02a574ed0b95506be0976c3fe49e96d4/q2_diversity/plugin_setup.py#L221-L361)

Feel free to ping us on Slack also!
